packages/backend/src/core/tx-count/BlockTxCountUpdater.ts

```diff
@@ -0,0 +1,85 @@
+import { Logger, TaskQueue } from '@l2beat/common'
+import { ProjectId, UnixTime } from '@l2beat/types'
+
+import { TxCountRepository } from '../../peripherals/database/TxCountRepository'
+import { EthereumClient } from '../../peripherals/ethereum/EthereumClient'
+import { Clock } from '../Clock'
+
+export class BlockTxCountUpdater {
+  private updateQueue = new TaskQueue<void>(() => this.update(), this.logger)
+  private blockQueue = new TaskQueue(this.getBlock.bind(this), this.logger, {
+    workers: 100,
+  })
+  private queuedBlocks = new Set<number>()
```

Comment:
i think this implementation is acceptable for now but WDYT about adding this feature directly to TaskQueue? IMO it can be useful in other places too.

Reply:
@krzkaczor  Should I add this as a task to backlog?